media-gfx/gscan2pdf/gscan2pdf-2.12.2.ebuild

```diff
@@ -0,0 +1,112 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+DIST_TEST="do"
+
+inherit optfeature perl-module virtualx xdg-utils
+
+DESCRIPTION="Scan documents, perform OCR, produce PDFs and DjVus"
+HOMEPAGE="http://gscan2pdf.sourceforge.net/"
+SRC_URI="mirror://sourceforge/${PN}/${P}.tar.xz"
+
+LICENSE="GPL-3"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="test"
+RESTRICT="!test? ( test )"
+
+RDEPEND="
+	dev-perl/Config-General
+	dev-perl/Data-UUID
+	dev-perl/Date-Calc
+	dev-perl/Filesys-Df
+	dev-perl/glib-perl
+	dev-perl/GooCanvas2
+	dev-perl/Gtk3
+	dev-perl/Gtk3-ImageView
+	dev-perl/Gtk3-SimpleList
+	dev-perl/HTML-Parser
+	dev-perl/Image-Sane
+	dev-perl/List-MoreUtils
+	dev-perl/Locale-Codes
+	dev-perl/Locale-gettext
+	dev-perl/Log-Log4perl
+	>=dev-perl/PDF-Builder-3.23.0
+	dev-perl/Proc-ProcessTable
+	dev-perl/Readonly
+	dev-perl/Set-IntSpan
+	dev-perl/Try-Tiny
+	virtual/perl-Archive-Tar
+	virtual/perl-Carp
+	virtual/perl-Data-Dumper
+	virtual/perl-File-Temp
+	virtual/perl-Getopt-Long
+	virtual/perl-threads
+	virtual/perl-threads-shared
+	media-gfx/imagemagick[png,tiff,perl]
+	media-gfx/sane-backends
+	media-libs/tiff"
+
+BDEPEND="
+	test? (
+		${RDEPEND}
+		dev-perl/IPC-System-Simple
+		dev-perl/Sub-Override
+		media-libs/fontconfig
+
+		app-text/djvu[jpeg,tiff]
+		app-text/poppler[utils]
+		app-text/tesseract[-opencl,osd(+),png,tiff]
+		app-text/unpaper
+		media-gfx/imagemagick[djvu,jpeg,png,tiff,perl,postscript]
+		media-gfx/sane-backends[sane_backends_test]
+		media-gfx/sane-frontends
+	)"
+
+IDEPEND="dev-util/desktop-file-utils"
```

Comment:
This is actually not *really* needed and ebuilds normally don't add these deps unless needed for something else like tests.

xdg-utils is designed to do nothing if the command isn't found and that's actually fine because that means a cache update is not needed "yet" -- it will be updated later given cache is done "all at once" and will include this.

It's a simple example of what IDEPEND could be used for though, thus the note in the guide.

Reply:
I don't follow: xdg.eclass would add a DEPEND on desktop-file-utils and this update would always get done?

Reply:
These updates aren't package specific, i.e. when a package updates cache it updates any missing entries.

And if the tools aren't installed yet, then nothing is using this cache and updating it doesn't matter. Once it is, cache will be updated (and will include this package).

So xdg.eclass' DEPEND was never really relevant, and there's especially no need to go out of your way to add this dependency manually.

Reply:
Finally I get it. Thanks for pointing out and explaining. Removed in a new patch (can squash if preferred).